packages/astro/src/runtime/server/endpoint.ts

```diff
@@ -1,13 +1,16 @@
 import { bold } from 'kleur/colors';
-import type { APIContext, EndpointHandler } from '../../@types/astro.js';
 import { REROUTABLE_STATUS_CODES, REROUTE_DIRECTIVE_HEADER } from '../../core/constants.js';
 import { EndpointDidNotReturnAResponse } from '../../core/errors/errors-data.js';
 import { AstroError } from '../../core/errors/errors.js';
 import type { Logger } from '../../core/logger/core.js';
+import type { APIRoute } from '../../types/public/common.js';
+import type { APIContext } from '../../types/public/context.js';
 
 /** Renders an endpoint request to completion, returning the body. */
 export async function renderEndpoint(
-	mod: EndpointHandler,
+	mod: {
+		[method: string]: APIRoute;
```

Comment:
Type was fully internal to this line, figured there was no point to it being a separate type.